src/module.c

```diff
@@ -40,6 +40,62 @@
 #define REDISTIMESERIES_GIT_SHA "unknown"
 #endif
 
+#define TIMESERIES_MODULE_ACL_CATEGORY_NAME "timeseries"
+
+int asprintf(char **str, const char *format, ...) {
+    va_list args;
+    va_start(args, format);
+
+    // First, determine the size of the formatted string
+    int size = vsnprintf(NULL, 0, format, args);
+    if (size < 0) {
+        va_end(args);
+        return -1;
+    }
+
+    // Allocate memory for the string
+    *str = (char *)malloc(size + 1);
+    if (!*str) {
+        va_end(args);
+        return -1;
+    }
+
+    // Write the formatted string into the allocated memory
+    va_start(args, format); // Restart the args for the second pass
+    int result = vsnprintf(*str, size + 1, format, args);
+    va_end(args);
+
+    if (result < 0) {
+        free(*str);
+        *str = NULL;
+    }
+
+    return result;
+}
+
+#define SetCommandAcls(ctx, cmd, acls)                                                             \
+    {                                                                                              \
+        RedisModuleCommand *command = RedisModule_GetCommand(ctx, cmd);                            \
+        if (command == NULL) {                                                                     \
+            return REDISMODULE_ERR;                                                                \
+        }                                                                                          \
+                                                                                                   \
+        char *categories = NULL;                                                                   \
+        if (!strcmp(acls, "")) {                                                                   \
+            categories = TIMESERIES_MODULE_ACL_CATEGORY_NAME;                                      \
+        } else {                                                                                   \
+            asprintf(&categories, "%s %s", acls, TIMESERIES_MODULE_ACL_CATEGORY_NAME);             \
+        }                                                                                          \
+                                                                                                   \
+        if (RedisModule_SetCommandACLCategories(command, categories) != REDISMODULE_OK) {          \
+            return REDISMODULE_ERR;                                                                \
+        }                                                                                          \
+    }
+
+#define RegisterCommandWithModesAndAcls(ctx, cmd, f, mode, acls)                                   \
+    __rmutil_register_cmd(ctx, cmd, f, mode);                                                      \
```

Comment:
i'm not gonna lie, i think the macros in `rmutil` make the code *less* readable. quite a lot of them could and should be functions.

for what it's worth, the `__` in this identifier makes this code potentially IFNDR.

Reply:
I fully agree. I just don't know if I should do anything about it now for this particular goal, or rather have a proper refactoring sitdown.

Reply:
i think i've brought this point up to multiple people, eg meir, guy, etc. it's always kicked down the road. "for a future pr" :disappointed: